source/base/polynomials_vector_anisotropic.cc

```diff
@@ -0,0 +1,301 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2005 - 2024 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+#include <deal.II/base/polynomials_vector_anisotropic.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/thread_management.h>
+
+#include <iomanip>
+#include <iostream>
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace
+{
+  // Create nodal anisotropic polynomials as the tensor product of Lagrange
+  // polynomials on Gauss-Lobatto points of the given degrees in the normal and
+  // tangential directions, respectively (we could also choose Lagrange
+  // polynomials on Gauss points but those are slightly more expensive to handle
+  // in classes).
+  std::vector<std::vector<Polynomials::Polynomial<double>>>
+  create_aniso_polynomials(const unsigned int dim,
+                           const unsigned int normal_degree,
+                           const unsigned int tangential_degree)
+  {
+    std::vector<std::vector<Polynomials::Polynomial<double>>> pols(dim);
+    if (normal_degree > 0)
+      pols[0] = Polynomials::generate_complete_Lagrange_basis(
+        QGaussLobatto<1>(normal_degree + 1).get_points());
+    else
+      pols[0] = Polynomials::generate_complete_Lagrange_basis(
+        QMidpoint<1>().get_points());
+    if (tangential_degree > 0)
+      for (unsigned int d = 1; d < dim; ++d)
+        pols[d] = Polynomials::generate_complete_Lagrange_basis(
+          QGaussLobatto<1>(tangential_degree + 1).get_points());
+    else
+      for (unsigned int d = 1; d < dim; ++d)
+        pols[d] = Polynomials::generate_complete_Lagrange_basis(
+          QMidpoint<1>().get_points());
```

Comment:
This the same point distribution as used in the function `PolynomialsVectorAnisotropic<dim>::get_polynomial_support_points()` below, isn't it? To reduce redundancy (and chances for mistakes if one were the change the point distribution), I suggest to add an additional function `std::vector<std::vector<Point<1>> create_anisotropic_support_points(dim, normal_degree, tangential_degree)` and use that function both for the `create_aniso_polynomials` here and in the function below.